Rebrand to Sokol Flex OS

kergoth opened the pull request:
Based upon #1, merge that first.

abelal commented on meta-sokol-flex-distro/conf/distro/sokol-flex.conf:
```diff
@@ -94,16 +91,6 @@ APPEND_KGDBOC = "kgdbwait kgdboc=${CMD_SERIAL_CONSOLE}"
 
 # Splash screen
 SPLASH:mel ?= "psplash"
```
still using mel here?

abelal replied:
there's also SDK_DEPLOY:mel in the same file.

kergoth replied:
SDK_DEPLOY at least must use an override as it's defined later on and will overwrite any direct definition. I suspect that was the case for splash at some point too, but will check.

abelal commented:
I am seeing the following with these changes
```
$ egrep --exclude-dir=.git --exclude=*.patch -Hri "mel|mentor" .
./conf/local.conf.sample:IMAGE_FEATURES_DEVELOPMENT:append:feature-mel-bsp = " codebench-debug ssh-server-openssh tools-profile"
./setup-environment:        OPTIONALLAYERS="${OPTIONALLAYERS-mentor-private tracing-layer}"
./scripts/capdebuginfo:unset BUILDERR   # Path and filename of the MEL build error log file
./scripts/capdebuginfo:   if [ -d "${HOMEDIR}"/.mentor ]; then
./scripts/capdebuginfo:      cp -a "${HOMEDIR}"/.mentor/.registry "${__outputdir}"/host/mentor-registry
./scripts/setup-flex-builddir:EXTRAFLEXLAYERS="${EXTRAFLEXLAYERS-${EXTRAMELLAYERS-}}"
./scripts/setup-flex-builddir:All layers whose directory name is update-* or meta-mentor-cve, at toplevel or
./scripts/setup-flex-builddir:                 $FLEXDIR/meta-mentor-cve/conf/layer.conf; do
./scripts/setup-flex-builddir:    # Pick up qemu layers from meta-mentor-bsp as we do
./scripts/setup-flex-builddir:            OPTIONALLAYERS="$OPTIONALLAYERS mentor-bsp-$MACHINE"
./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:MAINTAINER = "Siemens Digital Industries Software <embedded_support@mentor.com>"
./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:SPLASH:mel ?= "psplash"
./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:BB_GIT_SHALLOW_REVS:pn-linux-mel ??= "${@'${LINUX_VERSION_TAG}' if '${LINUX_VERSION_TAG}' and '${LINUX_VERSION}' else ''}"
./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:INHERIT += "${@bb.utils.contains('BBFILE_COLLECTIONS', 'mentor-private', 'pdk-license', '', d)}"
./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:SDK_DEPLOY:mel = "${DEPLOY_DIR_SDK}"
./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:# If meta-mentor-private is available, pull in the populate-ade class
./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:ADE_IMAGE_CLASS = "${@bb.utils.contains('BBFILE_COLLECTIONS', 'mentor-private', 'populate_ade archive_ade_downloads', '', d)}"
./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:include conf/distro/include/qt5-mel.conf
./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:include conf/distro/include/mel-security.conf
./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:include conf/distro/include/mel-swupdate.conf
./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:include conf/distro/include/mel-mcf.conf
./meta-sokol-flex-distro/conf/layer.conf:LAYERRECOMMENDS_sokol-flex-distro = "sokol-flex-support sokol-flex-staging sourcery mentor-private tracing-layer"
./meta-sokol-flex-distro/openembedded-layer/recipes-support/lvm2/lvm2_%.bbappend:RREPLACES:${PN}-flex += "${PN}-mel"
./meta-sokol-flex-distro/openembedded-layer/recipes-support/lvm2/lvm2_%.bbappend:RCONFLICTS:${PN}-flex += "${PN}-mel"
./meta-sokol-flex-distro/openembedded-layer/recipes-connectivity/libwebsockets/libwebsockets_4.2.2.bbappend:SRC_URI:sokol-flex = "git://github.com/MentorEmbedded/libwebsockets.git;protocol=https;branch=v4.2-stable"
./meta-sokol-flex-distro/recipes-core/kbd/kbd_2.4.0.bbappend:SRC_URI:prepend:sokol-flex = "git://github.com/MentorEmbedded/kbd;branch=2.4;protocol=https "
Binary file ./meta-sokol-flex-distro/recipes-core/psplash/sokol-flex/flex.png matches
./meta-sokol-flex-distro/sokol-flex-support/recipes-core/meta/archive-release.bbappend:    """Any non-public github repo or url including a mentor domain
./meta-sokol-flex-distro/sokol-flex-support/recipes-core/meta/archive-release.bbappend:    if 'MentorEmbedded' in test_url:
./meta-sokol-flex-distro/sokol-flex-support/recipes-core/meta/archive-release.bbappend:    elif 'mentor.com' in test_url or 'mentorg.com' in test_url:
./meta-sokol-flex-distro/recipes-support/db/db_%.bbappend:SRC_URI:sokol-flex = "git://github.com/MentorEmbedded/libdb;branch=5.3;protocol=https"
./meta-sokol-flex-common/classes/flex-updates-check.bbclass:    ${FLEXDIR}/meta-mentor-cve \
```

I understand that we aren't renaming linux-mel/linux-firmware-mel at this time but all else should ideally be renamed? an example is this still using the feature-mel-bsp whereas the newly created meta-flex-ref-bsps already [renamed](https://github.com/MentorEmbedded/meta-flex-ref-bsps/blob/main/meta-flex-bsp-common/conf/layer.conf#L38) the feature to feature-flex-bsp.

Additionally our feature includes are being updated as well so
```
include conf/distro/include/qt5-mel.conf
include conf/distro/include/mel-security.conf
include conf/distro/include/mel-swupdate.conf
include conf/distro/include/mel-mcf.conf
```
would need to be renamed here as well to
```
include conf/distro/include/flex-qt6.conf
include conf/distro/include/flex-security.conf
include conf/distro/include/flex-swupdate.conf
include conf/distro/include/flex-mcf.conf
```

abelal commented:
@kergoth I believe this needs some additional modifications.

kergoth commented:
[**@abelal**](https://github.com/abelal) commented on [Jul 26, 2022, 12:03 AM MST](https://github.com/MentorEmbedded/meta-sokol-flex/pull/2#issuecomment-1195088471 "2022-07-26T07:03:30Z - Replied by Github Reply Comments"):
> I am seeing the following with these changes
> ```
> $ egrep --exclude-dir=.git --exclude=*.patch -Hri "mel|mentor" .
> ./conf/local.conf.sample:IMAGE_FEATURES_DEVELOPMENT:append:feature-mel-bsp = " codebench-debug ssh-server-openssh tools-profile"
> ./setup-environment:        OPTIONALLAYERS="${OPTIONALLAYERS-mentor-private tracing-layer}"
> ./scripts/capdebuginfo:unset BUILDERR   # Path and filename of the MEL build error log file
> ./scripts/capdebuginfo:   if [ -d "${HOMEDIR}"/.mentor ]; then
> ./scripts/capdebuginfo:      cp -a "${HOMEDIR}"/.mentor/.registry "${__outputdir}"/host/mentor-registry
> ./scripts/setup-flex-builddir:EXTRAFLEXLAYERS="${EXTRAFLEXLAYERS-${EXTRAMELLAYERS-}}"
> ./scripts/setup-flex-builddir:All layers whose directory name is update-* or meta-mentor-cve, at toplevel or
> ./scripts/setup-flex-builddir:                 $FLEXDIR/meta-mentor-cve/conf/layer.conf; do
> ./scripts/setup-flex-builddir:    # Pick up qemu layers from meta-mentor-bsp as we do
> ./scripts/setup-flex-builddir:            OPTIONALLAYERS="$OPTIONALLAYERS mentor-bsp-$MACHINE"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:MAINTAINER = "Siemens Digital Industries Software <embedded_support@mentor.com>"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:SPLASH:mel ?= "psplash"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:BB_GIT_SHALLOW_REVS:pn-linux-mel ??= "${@'${LINUX_VERSION_TAG}' if '${LINUX_VERSION_TAG}' and '${LINUX_VERSION}' else ''}"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:INHERIT += "${@bb.utils.contains('BBFILE_COLLECTIONS', 'mentor-private', 'pdk-license', '', d)}"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:SDK_DEPLOY:mel = "${DEPLOY_DIR_SDK}"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:# If meta-mentor-private is available, pull in the populate-ade class
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:ADE_IMAGE_CLASS = "${@bb.utils.contains('BBFILE_COLLECTIONS', 'mentor-private', 'populate_ade archive_ade_downloads', '', d)}"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:include conf/distro/include/qt5-mel.conf
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:include conf/distro/include/mel-security.conf
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:include conf/distro/include/mel-swupdate.conf
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:include conf/distro/include/mel-mcf.conf
> ./meta-sokol-flex-distro/conf/layer.conf:LAYERRECOMMENDS_sokol-flex-distro = "sokol-flex-support sokol-flex-staging sourcery mentor-private tracing-layer"
> ./meta-sokol-flex-distro/openembedded-layer/recipes-support/lvm2/lvm2_%.bbappend:RREPLACES:${PN}-flex += "${PN}-mel"
> ./meta-sokol-flex-distro/openembedded-layer/recipes-support/lvm2/lvm2_%.bbappend:RCONFLICTS:${PN}-flex += "${PN}-mel"
> ./meta-sokol-flex-distro/openembedded-layer/recipes-connectivity/libwebsockets/libwebsockets_4.2.2.bbappend:SRC_URI:sokol-flex = "git://github.com/MentorEmbedded/libwebsockets.git;protocol=https;branch=v4.2-stable"
> ./meta-sokol-flex-distro/recipes-core/kbd/kbd_2.4.0.bbappend:SRC_URI:prepend:sokol-flex = "git://github.com/MentorEmbedded/kbd;branch=2.4;protocol=https "
> Binary file ./meta-sokol-flex-distro/recipes-core/psplash/sokol-flex/flex.png matches
> ./meta-sokol-flex-distro/sokol-flex-support/recipes-core/meta/archive-release.bbappend:    """Any non-public github repo or url including a mentor domain
> ./meta-sokol-flex-distro/sokol-flex-support/recipes-core/meta/archive-release.bbappend:    if 'MentorEmbedded' in test_url:
> ./meta-sokol-flex-distro/sokol-flex-support/recipes-core/meta/archive-release.bbappend:    elif 'mentor.com' in test_url or 'mentorg.com' in test_url:
> ./meta-sokol-flex-distro/recipes-support/db/db_%.bbappend:SRC_URI:sokol-flex = "git://github.com/MentorEmbedded/libdb;branch=5.3;protocol=https"
> ./meta-sokol-flex-common/classes/flex-updates-check.bbclass:    ${FLEXDIR}/meta-mentor-cve \
> ```
> 
> I understand that we aren't renaming linux-mel/linux-firmware-mel at this time but all else should ideally be renamed? an example is this still using the feature-mel-bsp whereas the newly created meta-flex-ref-bsps already [renamed](https://github.com/MentorEmbedded/meta-flex-ref-bsps/blob/main/meta-flex-bsp-common/conf/layer.conf#L38) the feature to feature-flex-bsp.
> 
> Additionally our feature includes are being updated as well so
> ```
> include conf/distro/include/qt5-mel.conf
> include conf/distro/include/mel-security.conf
> include conf/distro/include/mel-swupdate.conf
> include conf/distro/include/mel-mcf.conf
> ```
> would need to be renamed here as well to
> ```
> include conf/distro/include/flex-qt6.conf
> include conf/distro/include/flex-security.conf
> include conf/distro/include/flex-swupdate.conf
> include conf/distro/include/flex-mcf.conf
> ```

Yes, this is what I was referring to on the call, adapting to the renames happening in other layers which have not yet actually occurred. I was intending to add this to this PR today.

kergoth commented:
[**@abelal**](https://github.com/abelal) commented on [Jul 26, 2022, 12:03 AM MST](https://github.com/MentorEmbedded/meta-sokol-flex/pull/2#issuecomment-1195088471 "2022-07-26T07:03:30Z - Replied by Github Reply Comments"):
> I am seeing the following with these changes
> ```
> $ egrep --exclude-dir=.git --exclude=*.patch -Hri "mel|mentor" .
> ./conf/local.conf.sample:IMAGE_FEATURES_DEVELOPMENT:append:feature-mel-bsp = " codebench-debug ssh-server-openssh tools-profile"
> ./setup-environment:        OPTIONALLAYERS="${OPTIONALLAYERS-mentor-private tracing-layer}"
> ./scripts/capdebuginfo:unset BUILDERR   # Path and filename of the MEL build error log file
> ./scripts/capdebuginfo:   if [ -d "${HOMEDIR}"/.mentor ]; then
> ./scripts/capdebuginfo:      cp -a "${HOMEDIR}"/.mentor/.registry "${__outputdir}"/host/mentor-registry
> ./scripts/setup-flex-builddir:EXTRAFLEXLAYERS="${EXTRAFLEXLAYERS-${EXTRAMELLAYERS-}}"
> ./scripts/setup-flex-builddir:All layers whose directory name is update-* or meta-mentor-cve, at toplevel or
> ./scripts/setup-flex-builddir:                 $FLEXDIR/meta-mentor-cve/conf/layer.conf; do
> ./scripts/setup-flex-builddir:    # Pick up qemu layers from meta-mentor-bsp as we do
> ./scripts/setup-flex-builddir:            OPTIONALLAYERS="$OPTIONALLAYERS mentor-bsp-$MACHINE"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:MAINTAINER = "Siemens Digital Industries Software <embedded_support@mentor.com>"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:SPLASH:mel ?= "psplash"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:BB_GIT_SHALLOW_REVS:pn-linux-mel ??= "${@'${LINUX_VERSION_TAG}' if '${LINUX_VERSION_TAG}' and '${LINUX_VERSION}' else ''}"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:INHERIT += "${@bb.utils.contains('BBFILE_COLLECTIONS', 'mentor-private', 'pdk-license', '', d)}"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:SDK_DEPLOY:mel = "${DEPLOY_DIR_SDK}"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:# If meta-mentor-private is available, pull in the populate-ade class
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:ADE_IMAGE_CLASS = "${@bb.utils.contains('BBFILE_COLLECTIONS', 'mentor-private', 'populate_ade archive_ade_downloads', '', d)}"
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:include conf/distro/include/qt5-mel.conf
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:include conf/distro/include/mel-security.conf
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:include conf/distro/include/mel-swupdate.conf
> ./meta-sokol-flex-distro/conf/distro/sokol-flex.conf:include conf/distro/include/mel-mcf.conf
> ./meta-sokol-flex-distro/conf/layer.conf:LAYERRECOMMENDS_sokol-flex-distro = "sokol-flex-support sokol-flex-staging sourcery mentor-private tracing-layer"
> ./meta-sokol-flex-distro/openembedded-layer/recipes-support/lvm2/lvm2_%.bbappend:RREPLACES:${PN}-flex += "${PN}-mel"
> ./meta-sokol-flex-distro/openembedded-layer/recipes-support/lvm2/lvm2_%.bbappend:RCONFLICTS:${PN}-flex += "${PN}-mel"
> ./meta-sokol-flex-distro/openembedded-layer/recipes-connectivity/libwebsockets/libwebsockets_4.2.2.bbappend:SRC_URI:sokol-flex = "git://github.com/MentorEmbedded/libwebsockets.git;protocol=https;branch=v4.2-stable"
> ./meta-sokol-flex-distro/recipes-core/kbd/kbd_2.4.0.bbappend:SRC_URI:prepend:sokol-flex = "git://github.com/MentorEmbedded/kbd;branch=2.4;protocol=https "
> Binary file ./meta-sokol-flex-distro/recipes-core/psplash/sokol-flex/flex.png matches
> ./meta-sokol-flex-distro/sokol-flex-support/recipes-core/meta/archive-release.bbappend:    """Any non-public github repo or url including a mentor domain
> ./meta-sokol-flex-distro/sokol-flex-support/recipes-core/meta/archive-release.bbappend:    if 'MentorEmbedded' in test_url:
> ./meta-sokol-flex-distro/sokol-flex-support/recipes-core/meta/archive-release.bbappend:    elif 'mentor.com' in test_url or 'mentorg.com' in test_url:
> ./meta-sokol-flex-distro/recipes-support/db/db_%.bbappend:SRC_URI:sokol-flex = "git://github.com/MentorEmbedded/libdb;branch=5.3;protocol=https"
> ./meta-sokol-flex-common/classes/flex-updates-check.bbclass:    ${FLEXDIR}/meta-mentor-cve \
> ```
> 
> I understand that we aren't renaming linux-mel/linux-firmware-mel at this time but all else should ideally be renamed? an example is this still using the feature-mel-bsp whereas the newly created meta-flex-ref-bsps already [renamed](https://github.com/MentorEmbedded/meta-flex-ref-bsps/blob/main/meta-flex-bsp-common/conf/layer.conf#L38) the feature to feature-flex-bsp.
> 
> Additionally our feature includes are being updated as well so
> ```
> include conf/distro/include/qt5-mel.conf
> include conf/distro/include/mel-security.conf
> include conf/distro/include/mel-swupdate.conf
> include conf/distro/include/mel-mcf.conf
> ```
> would need to be renamed here as well to
> ```
> include conf/distro/include/flex-qt6.conf
> include conf/distro/include/flex-security.conf
> include conf/distro/include/flex-swupdate.conf
> include conf/distro/include/flex-mcf.conf
> ```

Also, that grep refers to multiple instances that should not be replaced, such as MentorEmbedded references on github, mentor.com references (stash) in layer git handling, etc. Once these references to files and features from other layers are corrected, the rest should be left as is.

kergoth commented:
[**@abelal**](https://github.com/abelal) commented on [Jul 26, 2022, 12:04 AM MST](https://github.com/MentorEmbedded/meta-sokol-flex/pull/2#issuecomment-1195089098 "2022-07-26T07:04:08Z - Replied by Github Reply Comments"):
> [@kergoth](https://github.com/kergoth) I believe this needs some additional modifications.

Updated. Interdiff:
```diff
 1:  387736f3 =  1:  ce355429 local.conf.sample: update archiver fragment from oe-core sample
 2:  45889e22 =  2:  aa5077e3 Rename layers to sokol-flex
 3:  06142a01 =  3:  bb39152f Rename mel distro to sokol-flex
 4:  d0a7d20d =  4:  05c2bd93 sokol-flex: update the HOME_URL
 5:  09c62bb4 =  5:  46656857 sokol-flex: remove unnecessary ADE_DISTRO
 6:  ebba771f =  6:  3c95a941 Mentor Embedded Linux Flex OS -> Sokol Flex OS
 7:  e1ef0440 =  7:  23712f5c MEL -> Sokol Flex OS
 8:  57306550 !  8:  568343a3 Rename/replace mel with flex
    @@ Commit message
         Signed-off-by: Christopher Larson <chris_larson@mentor.com>
     
      ## conf/local.conf.sample ##
    +@@ conf/local.conf.sample: MACHINE ??= "qemux86"
    + # details.
    + 
    + IMAGE_FEATURES_DEVELOPMENT ?= "debug-tweaks"
    +-IMAGE_FEATURES_DEVELOPMENT:append:feature-mel-bsp = " codebench-debug ssh-server-openssh tools-profile"
    ++IMAGE_FEATURES_DEVELOPMENT:append:feature-flex-bsp = " codebench-debug ssh-server-openssh tools-profile"
    + EXTRA_IMAGE_FEATURES = "${IMAGE_FEATURES_DEVELOPMENT} multilib-runtime"
    + 
    + # Image features for production-image
     @@ conf/local.conf.sample: BINUTILS_EXCEPTION = "${@'binutils:GPL-3.0-only' if d.getVar('PN') == 'binutils'
      #SOURCE_MIRROR_URL = "http://myserver/sources"
      
    @@ meta-sokol-flex-distro/conf/distro/sokol-flex.conf: TCMODE = "${@bb.utils.contai
      
      # Use a local PR server by default
      PRSERV_HOST ?= "localhost:0"
    +@@ meta-sokol-flex-distro/conf/distro/sokol-flex.conf: CMD_SERIAL_CONSOLE ?= "${@','.join(reversed('${SERIAL_CONSOLE}'.split()))}"
    + APPEND_KGDBOC = "kgdbwait kgdboc=${CMD_SERIAL_CONSOLE}"
    + 
    + # Splash screen
    +-SPLASH:mel ?= "psplash"
    ++SPLASH:sokol-flex ?= "psplash"
    + 
    + # Remove MACHINE from default volname
    + UBI_VOLNAME = "rootfs"
     @@ meta-sokol-flex-distro/conf/distro/sokol-flex.conf: SSTATE_MIRROR_SITES ?= ""
      SSTATE_MIRRORS += "${@''.join(r'file://.* %s/PATH;downloadfilename=PATH \n' % u for u in '${SSTATE_MIRROR_SITES}'.split())}"
      ## }}}1
    @@ meta-sokol-flex-distro/conf/distro/sokol-flex.conf: DISTRO_EXTRA_RRECOMMENDS +=
      
      # Ensure that the output directory always exists, this fixes use of meta-sca
      # with certain meta-imx recipes.
    +@@ meta-sokol-flex-distro/conf/distro/sokol-flex.conf: METADATA_REVISION := "${@base_detect_revision(d)}"
    + ## }}}1
    + ## SDK & Application Development Environment {{{1
    + # Use DEPLOY_DIR_ naming for consistency
    +-SDK_DEPLOY:mel = "${DEPLOY_DIR_SDK}"
    ++SDK_DEPLOY:sokol-flex = "${DEPLOY_DIR_SDK}"
    + DEPLOY_DIR_SDK ?= "${DEPLOY_DIR}/sdk"
    + 
    + # As we remove the toolchain from the sdk, naming it 'toolchain' is not
     @@ meta-sokol-flex-distro/conf/distro/sokol-flex.conf: IMAGE_CLASSES += "${ADE_IMAGE_CLASS}"
      ## Sokol Flex OS Releases {{{1
      # Default image for our installers
    @@ meta-sokol-flex-distro/conf/distro/sokol-flex.conf: FORKED_REPOS ?= "meta-virtua
      ## }}}1
      ## Includes {{{1
      
    -@@ meta-sokol-flex-distro/conf/distro/sokol-flex.conf: include conf/distro/include/mel-swupdate.conf
    - include conf/distro/include/mel-mcf.conf
    +@@ meta-sokol-flex-distro/conf/distro/sokol-flex.conf: require conf/distro/include/gdbserver-gplv3.inc
    + 
    + # QT5 configuration. These are includes rather than requires, as they live in
    + # other layers.
    +-include conf/distro/include/qt5-mel.conf
    ++include conf/distro/include/qt5-flex.conf
    + include conf/distro/include/qt5-versions.inc
    + 
    + # Security configuration.
    +-include conf/distro/include/mel-security.conf
    ++include conf/distro/include/flex-security.conf
    + 
    + # SWUpdate configuration.
    +-include conf/distro/include/mel-swupdate.conf
    ++include conf/distro/include/flex-swupdate.conf
    + 
    + # MCF configuration
    +-include conf/distro/include/mel-mcf.conf
    ++include conf/distro/include/flex-mcf.conf
      
      # INITRAMFS
     -INITRAMFS_IMAGE ?= "mel-initramfs-image"
    @@ meta-sokol-flex-support/recipes-core/meta/archive-release/runqemu.in: if ( eval
     
      ## scripts/capdebuginfo ##
     @@
    + # kind, whether express or implied.
      
      # Variables
    - unset BUILDERR   # Path and filename of the MEL build error log file
    +-unset BUILDERR   # Path and filename of the MEL build error log file
     -unset MELDIR     # Root path of the Sokol Flex OS installation directory
    ++unset BUILDERR   # Path and filename of the flex build error log file
     +unset FLEXDIR     # Root path of the Sokol Flex OS installation directory
      
      HOMEDIR=${HOME}
 9:  a342a486 =  9:  2ef19683 archive-release-downloads: Mentor -> Siemens
10:  573348ed = 10:  687d73ad Adjust for renamed repository (meta-mentor -> meta-sokol-flex)
11:  6fa18aa6 ! 11:  edda78b6 mentor-staging -> sokol-flex-staging
    @@ Metadata
     Author: Christopher Larson <chris_larson@mentor.com>
     
      ## Commit message ##
    -    mentor-staging -> sokol-flex-staging
    +    Replace references to mentor with sokol-flex/flex
     
         Signed-off-by: Christopher Larson <chris_larson@mentor.com>
     
    + ## meta-sokol-flex-common/classes/mentor-updates-check.bbclass => meta-sokol-flex-common/classes/flex-updates-check.bbclass ##
    +@@
    + UPDATES_CHECK_LAYERS ?= "\
    +     ${FLEXDIR}/updates/update-*/ \
    +     ${FLEXDIR}/update-*/ \
    +-    ${FLEXDIR}/meta-mentor-cve \
    ++    ${FLEXDIR}/meta-flex-cve \
    + "
    + UPDATES_CHECK_LAYERS[type] = "list"
    + 
    +
      ## meta-sokol-flex-distro/conf/distro/sokol-flex.conf ##
     @@ meta-sokol-flex-distro/conf/distro/sokol-flex.conf: FLEX_DEFAULT_DISTRO_FEATURES += "${@bb.utils.contains('MACHINE_FEATURES', 'x11',
      DISTRO_FEATURES ?= "${DISTRO_FEATURES_DEFAULT} ${DISTRO_FEATURES_LIBC} ${FLEX_DEFAULT_DISTRO_FEATURES}"
    @@ meta-sokol-flex-distro/conf/distro/sokol-flex.conf: FLEX_DEFAULT_DISTRO_FEATURES
      
      # Since many embedded systems don't have non-root users, allow autospawn for
      # root as well.
    +@@ meta-sokol-flex-distro/conf/distro/sokol-flex.conf: INHERIT += "deploy-license-manifest"
    + 
    + # Warn if there are available update layers which are not included in the
    + # current configuration
    +-INHERIT += "mentor-updates-check"
    ++INHERIT += "flex-updates-check"
    + 
    + # Check PDK license
    +-INHERIT += "${@bb.utils.contains('BBFILE_COLLECTIONS', 'mentor-private', 'pdk-license', '', d)}"
    ++INHERIT += "${@bb.utils.contains('BBFILE_COLLECTIONS', 'flex-private', 'pdk-license', '', d)}"
    + 
    + # Warn the user and disable rootfs resizing for non-GPLv3 builds
    + INHERIT += "resize-rootfs-gplv3"
    +@@ meta-sokol-flex-distro/conf/distro/sokol-flex.conf: IMAGE_CLASSES += "archive_sdk_downloads"
    + # Create a 'latest' symlink for the SDK
    + IMAGE_CLASSES += "sdk_latest_link"
    + 
    +-# If meta-mentor-private is available, pull in the populate-ade class
    +-ADE_IMAGE_CLASS = "${@bb.utils.contains('BBFILE_COLLECTIONS', 'mentor-private', 'populate_ade archive_ade_downloads', '', d)}"
    ++# If meta-flex-private is available, pull in the populate-ade class
    ++ADE_IMAGE_CLASS = "${@bb.utils.contains('BBFILE_COLLECTIONS', 'flex-private', 'populate_ade archive_ade_downloads', '', d)}"
    + IMAGE_CLASSES += "${ADE_IMAGE_CLASS}"
    + ## }}}1
    + ## Sokol Flex OS Releases {{{1
    +
    + ## meta-sokol-flex-distro/conf/layer.conf ##
    +@@ meta-sokol-flex-distro/conf/layer.conf: BBFILE_PRIORITY_sokol-flex-distro = "12"
    + BBFILE_PATTERN_sokol-flex-distro = "^${LAYERDIR_RE}/"
    + LAYERSERIES_COMPAT_sokol-flex-distro = "kirkstone"
    + LAYERDEPENDS_sokol-flex-distro = "core sokol-flex-common"
    +-LAYERRECOMMENDS_sokol-flex-distro = "sokol-flex-support sokol-flex-staging sourcery mentor-private tracing-layer"
    ++LAYERRECOMMENDS_sokol-flex-distro = "sokol-flex-support sokol-flex-staging sourcery flex-private tracing-layer"
    + LAYERRECOMMENDS_sokol-flex-distro += "openembedded-layer filesystems-layer networking-layer multimedia-layer gplv2"
     
      ## meta-sokol-flex-staging/conf/layer.conf ##
     @@ meta-sokol-flex-staging/conf/layer.conf: LAYERSERIES_COMPAT_sokol-flex-staging = "kirkstone"
    @@ meta-sokol-flex-staging/security/recipes-security/scapy/python3-scapy_%.bbappend
     +RDEPENDS:${PN}:append:feature-sokol-flex-staging = "\
          ${PYTHON_PN}-fcntl \
      "
    +
    + ## scripts/setup-flex-builddir ##
    +@@ scripts/setup-flex-builddir: If the build directory already exists and is configured, you can leave MACHINE
    + unspecified, and the script will set you up for that build directory without
    + changing its configuration.
    + 
    +-All layers whose directory name is update-* or meta-mentor-cve, at toplevel or
    ++All layers whose directory name is update-* or meta-flex-cve, at toplevel or
    + a subdir, of the parent of meta-sokol-flex, will be automatically included, as
    + a mechanism for delivering sets of updates for releases.
    + 
    +@@ scripts/setup-flex-builddir: process_arguments () {
    +     updatelayers=
    +     for lconf in $FLEXDIR/update-*/conf/layer.conf \
    +                  $FLEXDIR/*/update-*/conf/layer.conf \
    +-                 $FLEXDIR/meta-mentor-cve/conf/layer.conf; do
    ++                 $FLEXDIR/meta-flex-cve/conf/layer.conf; do
    +         updatelayer="${lconf%/conf/layer.conf}"
    +         if [ -e "$updatelayer" ]; then
    +             updatelayers="$updatelayers $updatelayer"
    +@@ scripts/setup-flex-builddir: setup_builddir () {
    +         fi
    +     fi
    + 
    +-    # Pick up qemu layers from meta-mentor-bsp as we do
    ++    # Pick up qemu layers from meta-flex-bsp as we do
    +     # not create separate machines for them
    +     case ${MACHINE} in
    +         qemu*)
    +-            OPTIONALLAYERS="$OPTIONALLAYERS mentor-bsp-$MACHINE"
    ++            OPTIONALLAYERS="$OPTIONALLAYERS flex-bsp-$MACHINE"
    +             ;;
    +     esac
    + 
    +
    + ## setup-environment ##
    +@@ setup-environment: else
    +             fi
    +         done
    + 
    +-        OPTIONALLAYERS="${OPTIONALLAYERS-mentor-private tracing-layer}"
    ++        OPTIONALLAYERS="${OPTIONALLAYERS-flex-private tracing-layer}"
    +         # Customer directory layers handling (e.g. <customername>-custom)
    +         for layercheck in . $layerdir/..; do
    +             if [ -e "$layercheck/customer.conf" ]; then
12:  752f72d8 <  -:  -------- mentor-updates-check -> flex-updates-check
```

abelal commented on meta-sokol-flex-distro/openembedded-layer/recipes-support/lvm2/lvm2_%.bbappend:
```diff
@@ -4,16 +4,18 @@
+SYSTEMD_SERVICE:${PN}-flex = "lvm2-flex.service"
+SYSTEMD_AUTO_ENABLE:${PN}-flex = "enable"
+RREPLACES:${PN}-flex += "${PN}-mel"
+RCONFLICTS:${PN}-flex += "${PN}-mel"
```
why do we need to have the rreplaces and conflicts here? we've already made the transition to flex and haven't done a release on ginkgo/kirkstone that would use lvm2-mel

kergoth replied:
It's best practice in Yocto projects to allow for package upgrades to happen. Admittedly we don't do any testing with package feeds, and certainly don't test upgrades across major version, but there's no real cost to allowing it to work. I'm used to considering it when I work on recipes. I don't feel terribly strongly about this, however.

kergoth replied:
It's best practice in Yocto projects to allow for package upgrades to happen. Admittedly we don't do any testing with package feeds, and certainly don't test upgrades across major version, but there's no real cost to allowing it to work.. I'm used to considering it when I work on recipes. I don't feel terribly strongly about this, however.

Noor-Ahsan commented:
I am going to merge this
